pkg/token/cache.go

```diff
@@ -38,19 +42,43 @@ func (k CacheKey) String() string {
 	return fmt.Sprintf("{%s:role.%s,%s,%d,%d}", k.Domain, k.Role, k.ProxyForPrincipal, k.MinExpiry, k.MaxExpiry)
 }
 
+func (k CacheKey) Size() uint {
+	structSize := uint(unsafe.Sizeof(k))
+	// unsafe.Sizeof() ONLY count the string struct, need to count the actual string block explicitly
+	stringSize := len(k.Domain) + len(k.ProxyForPrincipal) + len(k.Role)
+	return structSize + uint(stringSize)
+}
+
 type LockedTokenCache struct {
 	cache map[CacheKey]Token
 	lock  sync.RWMutex
+
+	// memoryUsage is the estimated number of bytes used by the cache.
+	memoryUsage int64
 }
 
 func NewLockedTokenCache() *LockedTokenCache {
-	return &LockedTokenCache{cache: make(map[CacheKey]Token)}
+	return &LockedTokenCache{
+		cache:       make(map[CacheKey]Token),
+		memoryUsage: 0,
+	}
 }
 
 func (c *LockedTokenCache) Store(k CacheKey, t Token) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
+	oldToken, ok := c.cache[k]
```

Comment:
```suggestion
	oldToken, exists := c.cache[k]
```

Reply:
(Optional)

Reply:
go doc uses 

https://go.dev/blog/maps

Reply:
not fix